gensim/test/test_fasttext.py

```diff
@@ -1300,6 +1306,253 @@ def _run(self, fin):
+        model_params = {
+            "size": 10,
+            "min_count": 1,
+            "hs": 1,
+            "sg": 0,
+            "negative": 0,
+            "seed": 42,
+            "workers": 1}
+        self._check_roundtrip_file_file(model_params)
+
+
+def _save_test_model(out_base_fname, model_params, fasttext_cmd):
+    inp_fname = datapath('lee_background.cor')
+
+    model_type = "cbow" if model_params["sg"] == 0 else "skipgram"
+    size = str(model_params["size"])
+    seed = str(model_params["seed"])
+
+    cmd = fasttext_cmd + " " + model_type + " -input " + inp_fname + \
+            " -output " + out_base_fname + " -dim " + size + " -seed " + seed
+    subprocess.run(cmd, shell=True)
+
+
+class SaveFacebookFormatFileFastTextTest(unittest.TestCase):
+    """
+    This class containts tests that check the following scenario:
+
+    + create binary fastText file model1.bin using facebook_binary (FT)
+    + load file model1.bin to variable `model`
+    + save `model` to model2.bin using gensim
+    + check if files model1.bin and model2.bin are byte-identical
+
+    Requires env. variable FT_HOME to point to location of Facebook fastText binary
+    """
+
+    def _check_roundtrip_file_file(self, model_params):
+        ft_home = os.environ.get("FT_HOME", None)
+        fasttext_cmd = os.path.join(ft_home, "fasttext")
+
+        # fasttext tool creates both *vec and *bin files, so we have to remove both, even thought *vec is unused
+
+        with temporary_file("roundtrip_file_to_file1.bin") as fpath1bin, \
+            temporary_file("roundtrip_file_to_file2.bin") as fpath2bin, \
+            temporary_file("roundtrip_file_to_file1.vec") as fpath1vec:  # noqa:F841
+
+            fpath1base = fpath1bin[:-4]
+            _save_test_model(fpath1base, model_params, fasttext_cmd)
+            model = gensim.models.fasttext.load_facebook_model(fpath1bin)
+            gensim.models.fasttext.save_facebook_model(model, fpath2bin)
+            self.assertEqual(_read_binary_file(fpath1bin), _read_binary_file(fpath2bin))
+
+    @unittest.skipIf(not os.environ.get("FT_HOME", None), "FT_HOME env variable not set, skipping test")
+    def test_roundtrip_file_file_skipgram(self):
+        model_params = {"size": 10, "sg": 1, "seed": 42}
+        self._check_roundtrip_file_file(model_params)
+
+    @unittest.skipIf(not os.environ.get("FT_HOME", None), "FT_HOME env variable not set, skipping test")
+    def test_roundtrip_file_file_cbow(self):
+        model_params = {"size": 10, "sg": 0, "seed": 42}
+        self._check_roundtrip_file_file(model_params)
+
+
+def _parse_wordvectors(text):
+    def _conv_line_to_array(line):
+        return np.array([float(s) for s in line.split()[1:]], dtype=np.float32)
+
+    return np.array([_conv_line_to_array(l) for l in text.splitlines()], dtype=np.float32)
+
+
+def _read_wordvectors_using_fasttext(fasttext_cmd, fasttext_fname, words):
+    cmd = fasttext_cmd + " print-word-vectors " + fasttext_fname
```

Comment:
This is the [recommended way](https://docs.python.org/3.5/library/subprocess.html#popen-constructor) to pass the command:

```suggestion
    cmd = [fasttext_cmd, "print-word-vectors", fasttext_fname]
```

If you do it this way, then you don't need `shell=True` (unless you have another reason to use it, in that case, please let me know).

Reply:
Good point. Corrected.